Source/NSOrderedSet.m

```diff
@@ -0,0 +1,1911 @@
+
+/** Interface for NSOrderedSet, NSMutableOrderedSet for GNUStep
+   Copyright (C) 2019 Free Software Foundation, Inc.
+
+   Written by: Gregory John Casamento <greg.casamento@gmail.com>
+   Created: May 17 2019
+
+   This file is part of the GNUstep Base Library.
+
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2 of the License, or (at your option) any later version.
+
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Library General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02111 USA.
+
+*/
+#import "common.h"
+#import "Foundation/NSArray.h"
+#import "Foundation/NSAutoreleasePool.h"
+#import "Foundation/NSOrderedSet.h"
+#import "Foundation/NSCoder.h"
+#import "Foundation/NSArray.h"
+#import "Foundation/NSEnumerator.h"
+#import "Foundation/NSKeyValueCoding.h"
+#import "Foundation/NSValue.h"
+#import "Foundation/NSException.h"
+#import "Foundation/NSPredicate.h"
+#import "Foundation/NSLock.h"
+#import "Foundation/NSKeyValueObserving.h"
+
+#import <GNUstepBase/GSBlocks.h>
+#import "Foundation/NSKeyedArchiver.h"
+#import "GSPrivate.h"
+#import "GSFastEnumeration.h"
+#import "GSDispatch.h"
+#import "GSSorting.h"
+
+@class	GSOrderedSet;
+@interface GSOrderedSet : NSObject	// Help the compiler
+@end
+
+@class	GSMutableOrderedSet;
+@interface GSMutableOrderedSet : NSObject	// Help the compiler
+- (void) _raiseRangeExceptionWithIndex: (NSUInteger)index from: (SEL)sel;
+@end
+
+@interface NSMutableOrderedSet (Private)
+- (void) _insertObject: (id)object atIndex: (NSUInteger)index;
```

Comment:
This method is only implemented on GSMutableOrderedSet and if we use it here, this creates an undocumented  requirement on subclasses. We should try to come up with an implementation without this method.